training/views.py

```diff
@@ -0,0 +1,255 @@
+
+	return redirect(reverse('training_view', kwargs={'event_id': event_id}))
+
+@staff_required
+def admin(request):
+	'''Simple admin page for training module'''
+
+	template = 'training/admin/index.html'
+	context = {
+		'future_events': models.Event.objects.filter(date_time__gte=datetime.now()).order_by('-date_time'),
+		'past_events': models.Event.objects.filter(date_time__lt=datetime.now()).order_by('date_time'),
+		'training': models.Training.objects.all()
+	}
+
+	return render(request, template, context)
+
+@staff_required
+def new_training(request, training_id=None):
+	'''Staff can add a new training module'''
+
+	if training_id:
+		training = get_object_or_404(models.Training, pk=training_id)
+		form = forms.TrainingForm(instance=training)
+	else:
+		training = None
+		form = forms.TrainingForm()
+
+	if request.POST:
+		if training_id:
+			form = forms.TrainingForm(request.POST, instance=training)
+		else:
+			form = forms.TrainingForm(request.POST)
+
+		if form.is_valid():
+			form.save()
+
+		if training_id:
+			messages.add_message(request, messages.INFO, 'Scheme updated.')
+		else:
+			messages.add_message(request, messages.INFO, 'Scheme added.')
+
+		return redirect(reverse('training_admin'))
+
+
+	template = 'training/admin/new_training.html'
+	context = {
+		'form': form,
+		'training': training,
+	}
+
+	return render(request, template, context)
+
+@staff_required
+def new_event(request, event_id=None):
+	'''Staff can add a new event'''
+
+	if event_id:
+		event = get_object_or_404(models.Event, pk=event_id)
+		form = forms.EventForm(instance=event)
+	else:
+		event = None
+		form = forms.EventForm()
+
+	if request.POST:
+		if event_id:
+			form =  forms.EventForm(request.POST, instance=event)
+		else:
+			form = forms.EventForm(request.POST)
+
+		if form.is_valid():
+			form.save()
+
+			if event_id:
+				messages.add_message(request, messages.INFO, 'Event updated.')
+			else:
+				messages.add_message(request, messages.INFO, 'Event added.')
+
+			return redirect(reverse('training_admin'))
+
+	template = 'training/admin/new_event.html'
+	context = {
+		'form': form,
+		'event': event,
+	}
+
+	return render(request, template, context)
+
+@staff_required
+def delete_event(request, event_id):
+	'''Staff can delete an existing event'''
+
+	event = get_object_or_404(models.Event, pk=event_id)
+	event.delete()
+
+	return redirect(reverse('training_admin')) 
+
+@staff_required
+def attendance_register(request, event_id):
+	'''Staff can view a register of those signed up to an event'''
+
+	event = get_object_or_404(models.Event, pk=event_id)
+
+	template = 'training/admin/register.html'
+	context = {
+		'event': event,
+	}
+
+	return render(request, template, context)
+
+@staff_required
+def confirm_attendance(request, event_id, attendee_id):
+	'''Staff can confirm an attendee was at an event'''
+
+	attendee = get_object_or_404(models.Attendee, pk=attendee_id, event__pk=event_id)
+
+	attendee.confirmed = True
+	attendee.save()
```

Comment:
If you have time, it would be nice to track when the confirmation happened and which staff member performed the confirmation. It helps with auditing and reporting to have this kind of info in the database.